.github/workflows/documentation.yml

```diff
@@ -33,7 +39,7 @@ jobs:
     name: Deploy the documentation on Github pages
     runs-on: ubuntu-latest
     needs: generate
-    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    if: github.event_name == 'push'
```

Comment:
Why remove the master check ?

Reply:
Because currently we deploy from `develop`, and so I though it was better for the doc to be synced with develop.
Originally, I was keeping `master` for release available on `pypi.org`, not `test.pypi.org`. But this can change.

Reply:
No problem with either `master` or `develop`, but I think the workflow should only update the doc on `push on develop` otherwise we will update with draft content

Reply:
It is already the case as the workflow triggers only on `push` master & develop (other are PR), but I agree it is not very clear. I'll update the condition.